botocore/auth.py

```diff
@@ -230,7 +231,7 @@ def canonical_headers(self, headers_to_sign):
         for key in sorted_header_names:
             value = ','.join(self._header_value(v) for v in
                              sorted(headers_to_sign.get_all(key)))
-            headers.append('%s:%s' % (key, value))
+            headers.append('%s:%s' % (key, ensure_unicode(value)))
```

Comment:
Because we're tossing unicode into the headers. If we don't make sure it's encoded then we'll get ascii failures down the line.